exoscale/api/compute.py

```diff
@@ -2001,6 +2197,121 @@ def get_instance_type(self, name=None, id=None):
 
         return instance_types[0]
 
+    ### Instance Pool
+
+    def create_instance_pool(
+        self,
+        zone,
+        name,
+        size,
+        instance_type,
+        instance_template,
+        instance_root_disk_size=10,
+        instance_security_groups=None,
+        instance_private_networks=None,
+        instance_enable_ipv6=False,
+        instance_ssh_key=None,
+        instance_user_data=None,
+        description=None,
+    ):
+        """
+        Create a Compute Instance Pool.
+
+        Parameters:
+            zone (Zone): the zone in which to create the instance pool
+            name (str): the name of the instance pool
+            description (str): a description of the instance pool
+            size (int): the number of Compute instance members the pool must manage
+            instance_type (InstanceType): the Compute instance members type
+            instance_template (InstanceTemplate): the Compute instance template to use
+                for members
+            instance_root_disk_size (int): the Compute instance members root disk size
+                in GB
+            instance_private_networks ([PrivateNetwork]): a list of Private Networks to
+                attach the Compute instance members to
+            instance_security_groups ([SecurityGroup]): a list of Security Groups to
+                attach the Compute instance members to
+            instance_enable_ipv6 (bool): a flag indicating whether to enable IPv6 on
+                the Compute instance members public network interface
+            instance_ssh_key (SSHKey): a SSH Key to deploy on the Compute instance
+                members
+            instance_user_data (str): a cloud-init user data configuration to apply to
+                the Compute instance members
+
+        Returns:
+            InstancePool: the Compute Instance Pool created
+        """
+
+        if size < 0:
+            raise ValueError("size must be >= 0")
+
+        try:
+            res = self.cs.createInstancePool(
+                name=name,
+                description=description,
+                size=size,
+                zoneid=zone.id,
+                serviceofferingid=instance_type.id,
+                templateid=instance_template.id,
+                rootdisksize=instance_root_disk_size,
+                securitygroupids=list(i.id for i in instance_security_groups)
+                if instance_security_groups
+                else None,
+                networkids=list(i.id for i in instance_private_networks)
```

Comment:
I take your point but I'll make the related change in another PR because there's the same pattern elsewhere throughout the library, I'd prefer to change those in a separate context if that's alright with you.